14_functions_lab/01.py

```diff
@@ -1,9 +1,21 @@
-import unittest
+#!/usr/bin/python
+def mysum(*nums):
+	s = 0
+	for i in nums:
+		try:
+			s += int(i)
+		except:
+			pass
+	return s
+	
 
-"""
-Write 2 functions:
-    mysum - returns the sum of its input arguments
-    mymul - returns the multiplication of its input arguments
-    Ignore non-numeric arguments
-"""
+def mymul(*nums):
+	s = 1
+	for i in nums:
+		try:
+			s = s * int(i)
+		except:
+			pass
+	return s	
 
```

Comment:
Both works, however I'd write a helper function here (and now you can easily implement more functions in this style without copy/pasting):

```
def reduce_ignoring_exceptions(op, initial, *items):
    acc = initial
    for val in items:
        try:
            acc = op(acc, val)
        except:
            pass

    return acc

mysum = lambda *items: reduce_ignoring_exceptions(lambda a, b: a + int(b), 0, *items)

mymul = lambda *items: reduce_ignoring_exceptions(lambda a, b: a * int(b), 1, *items)

print mysum(10, 'a', 'b', 20)

print mymul(10, 'a', 'b', 20)
 
```